src/main/java/com/devonfw/ide/sonarqube/common/impl/check/DevonNamingConventionClassExtendsClassCtoCheck.java

```diff
@@ -0,0 +1,24 @@
+package com.devonfw.ide.sonarqube.common.impl.check;
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+
+/**
+ * @author vhacimuf
+ *
+ */
+
+@Rule(key = "Devon4j:S2", name = "Devon Naming Convention Check", //
```

Comment:
Just as a first feedback:
the `@Rule` `key` has to be unique per rule.
In sonar-devon-plugin we defined a letter for each ruletype:
* S = scope
* C = component
* L = layer
* P = package
* E = external/extension (3rd party)
So you could add 
* N = naming

After that comes a sequential number so the key is unique within the plugin. The Prefix "Devon4j:" ensures that it is also unique for the entire sonarqube ecosystem.

Reply:
In my last commit I assigned unique keys